src/compiler/crystal/macros/methods.cr

```diff
@@ -156,6 +158,27 @@ module Crystal
       end
     end
 
+    def interpret_parse_type(node)
+      interpret_check_args_toplevel do |arg|
+        arg.accept self
+        type_name = case last = @last
+                    when StringLiteral then last.value
+                    else
+                      arg.raise "argument to parse_type must be a StringLiteral, not #{last.class_desc}"
+                    end
+
+        arg.raise "argument to parse_type cannot be an empty value" if type_name.blank?
+
+        parser = Crystal::Parser.new type_name
+        parser.next_token_skip_statement_end
```

Comment:
I think this means that if you give it ";;;Foo", it will parse to "Foo". Is there a need to skip those? I feel like this could just be `next_token`.